### PR TITLE
add Stack.drop

### PR DESCRIPTION
**Pull request by redianthus**
Hi,

This PR adds a `val drop : 'a t -> unit` function to the `Stack` module.

It behaves as `Stack.pop` except that it returns `()` instead of the topmost element.

This is useful when you're e.g. writing a WebAssembly interpreter. :)

More seriously, Sherlocode shows [118 results] when searching for `ignore (Stack.pop` and there are [probably more] - IIRC Sherlocode doesn't index all opam packages so I bet the real number is higher.

With the feature freeze, I wasn't sure about when this can be merged. I used `@since 5.00`, let me know if this should be changed.

[118 results]: https://sherlocode.com/?q=ignore%20(Stack.pop
[probably more]: https://sherlocode.com/?q=ignore%20%40%40%20Stack.pop

### CODE REVIEW

**Review by gasche — APPROVED** [review]
It's a useful proposal, I'm Approving, but indeed that looks like post-freeze material.

(I'm not sure whether the ice age will thaw off before or after the 5.00 release, so I just added a `post-freeze` milestone.)

**Comment by redianthus** [comment]
ping @gasche, I rebased on trunk and I believe it can be merged :)

**Comment by gasche** [comment]
We require stdlib changes to be approved by two maintainers instead of one.(cc  @Octachron, @nojb)

**Comment by redianthus** [comment]
I changed `@since 5.0` to `@since 5.1.0`.

**Review by nojb — APPROVED** [review]
LGTM

I am wondering though if `drop : 'a t -> int -> unit` may be more generally useful ?

File: Changes
Comment (nojb) [inline Changes:18]: There is a missing empty line separating the two entries.

**Comment by redianthus** [comment]
@nojb, fixed !

> I am wondering though if drop : 'a t -> int -> unit may be more generally useful ?

From the result I get on sherlocode, people seem to drop a single element most of the time.

**Comment by xavierleroy** [comment]
> From the result I get on sherlocode, people seem to drop a single element most of the time.

Agreed.  The current API gives no facilities for pushing N elements at a time or popping N elements at a time, so dropping N elements at a time doesn't make much sense to me.

**Comment by nojb** [comment]
> @nojb, fixed !
> 
> > I am wondering though if drop : 'a t -> int -> unit may be more generally useful ?
> 
> From the result I get on sherlocode, people seem to drop a single element most of the time.

Makes sense, the PR diff looks good and it has two approvals. Merging.

**Pull request merged**
